### ISSUE TITLE
[Outlines Support] Enhance With Structured Text Generation when VLLM is used

### ISSUE THREAD

**Issue by dprokhorov17** [langchain4j/langchain4j#1501]
Would love to see an integration of the outlines library (https://github.com/outlines-dev/outlines)

**Comment by langchain4j** [comment]
Hi @dprokhorov17 what features of outlines do you mean?

**Comment by dprokhorov17** [comment]
Hi @langchain4j . I mean especially the Extra Parameters of the Chat API: https://docs.vllm.ai/en/latest/serving/openai_compatible_server.html#extra-parameters-for-chat-api

![image](https://github.com/user-attachments/assets/5c9e08fe-5664-430b-97c6-545acca6e2ab)

I think, that this should not be that difficult to implement...

**Comment by langchain4j** [comment]
@dprokhorov17 so this is vLLM-specific, not outlines, right?

**Comment by dprokhorov17** [comment]
@langchain4j Yes, you're right.

**Comment by langchain4j** [comment]
@dprokhorov17 we do not have vLLM-specific integration, how do you see this implemented?

**Comment by dprokhorov17** [comment]
What do you need a vLLM-specific intergration? You have the OpenAiChatModel that is compatible to the vLLM OpenAI server:

```
OpenAiChatModel model = OpenAiChatModel
        .builder()
        .modelName("meta-llama/Meta-Llama-3-70B-Instruct")
        .baseUrl("http://my.ip-address:8000/v1")
        .apiKey("myToken")
        .temperature(0.0)
        .maxTokens(1000)
        .build();
```

**Comment by langchain4j** [comment]
@dprokhorov17 correct, so how do you see this being implemented with `OpenAiChatModel`? It would be great to keep OpenAI integration as vLLM-agnostic as possible
